lib/chat/db/scope/key_scope.ex

```diff
@@ -0,0 +1,184 @@
+defmodule Chat.Db.Scope.KeyScope do
+  @moduledoc """
+  Builds db keys accessible by keys_list
+  """
+
+  alias Chat.Utils
+
+  def get_keys(db, pub_keys_list) do
+    hashes = hash_pubkeys(pub_keys_list)
+
+    CubDB.with_snapshot(db, fn snap ->
+      MapSet.new()
+      |> add_full_users(snap)
+      |> add_dialogs(snap, hashes)
+      |> add_rooms(snap, hashes)
+      |> add_content(snap, hashes)
+      |> MapSet.to_list()
+    end)
+  end
+
+  defp add_full_users(acc_set, snap) do
+    snap
+    |> db_keys_stream({:users, 0}, {:"users\0", 0})
+    |> union_set(acc_set)
+  end
+
+  defp add_dialogs(acc_set, snap, {_binhashes, hashes}) do
+    dialog_keys =
+      snap
+      |> db_keys_stream({:dialogs, 0}, {:"dialogs\0", 0})
+      |> Stream.filter(fn {:dialogs, <<one::binary-size(64), other::binary-size(64)>>} ->
+        MapSet.member?(hashes, one) or MapSet.member?(hashes, other)
+      end)
+      |> MapSet.new()
+
+    dialog_binhashes =
+      dialog_keys
+      |> Enum.map(fn {:dialogs, hash} ->
+        hash
+        |> Utils.hash()
+        |> Utils.binhash()
+      end)
+      |> MapSet.new()
+
+    snap
+    |> db_keys_stream({:dialog_message, 0, 0, 0}, {:"dialog_message\0", 0, 0, 0})
+    |> Stream.filter(fn {:dialog_message, binhash, _, _} ->
+      MapSet.member?(dialog_binhashes, binhash)
+    end)
+    |> union_set(dialog_keys)
+    |> union_set(acc_set)
+  end
+
+  defp add_rooms(acc_set, snap, {binhashes, hashes}) do
+    with_rooms =
+      snap
+      |> db_keys_stream({:rooms, 0}, {:"rooms\0", 0})
+      |> Stream.filter(fn {:rooms, hash} ->
+        MapSet.member?(hashes, hash)
+      end)
+      |> union_set(acc_set)
+
+    snap
+    |> db_keys_stream({:room_message, 0, 0, 0}, {:"room_message\0", 0, 0, 0})
+    |> Stream.filter(fn {:room_message, room_binhash, _, _} ->
+      MapSet.member?(binhashes, room_binhash)
+    end)
+    |> union_set(with_rooms)
+  end
+
+  defp add_content(acc_set, snap, {_, hashes}) do
+    file_index =
+      snap
+      |> db_keys_stream({:file_index, 0, 0, 0}, {:"file_index\0", 0, 0, 0})
+      |> Stream.filter(fn {:file_index, reader_hash, _file_key, _message_id} ->
+        MapSet.member?(hashes, reader_hash)
+      end)
+      |> MapSet.new()
+
+    file_keys =
+      file_index
+      |> Enum.map(fn {:file_index, _reader_hash, file_key, _message_id} -> file_key end)
+      |> MapSet.new()
+
+    file_chunks =
+      snap
+      |> db_keys_stream({:file_chunk, 0, 0, 0}, {:"file_chunk\0", 0, 0, 0})
+      |> Stream.filter(fn {:file_chunk, file_key, _chunk_start, _chunk_end} ->
+        MapSet.member?(file_keys, file_key)
+      end)
+      |> MapSet.new()
```

Comment:
db should not have `file_chunks` by now. 
But we need to produce them for files (from fileFs) to be copied.
We can do so from `chunk_keys`

Reply:
That's right. I realized that while I was writing the tests that it's just a convenient API for the file system and that there's no DB for that. I'll remove this part of the code.